experimental/iterators/include/iterators/Dialect/Iterators/IR/IteratorsOps.td

```diff
@@ -50,6 +50,33 @@ def Iterators_PrintOp : Iterators_Base_Op<"print"> {
   let arguments = (ins Iterators_LLVMStructOfNumerics:$element);
 }
 
+def Iterators_ColumnarBatchFromMemrefsOp
+    : Iterators_Base_Op<"columnar_batch_from_memrefs"> {
```

Comment:
How about "view_as_columnar_batch" ?
This is more consistent with the fact that there is no underlying alloc.
The from_memref is visible in the type so is redundant.
Additionally, in the future, you may also want to allow tensors.

Reply:
I agree with the `from_memrefs` part.

I am not sure about `view`. It's not a view in the sense of `ViewLikeOpInterface`. Also, other ops also don't make their view intensions explicit (such as [Vector_TypeCast](https://github.com/llvm/llvm-project/blob/b83d0d/mlir/include/mlir/Dialect/Vector/IR/VectorOps.td#L2098)).

So maybe just `to_columnar_batch` (like `to_tensor` and `to_memref` in bufferization)?

I thought about supporting tensors as well (and had some hacky version implemented at some point) but then didn't see the benefit compared to converting manually with `bufferization.to_memref`. Do you see one? (And how'd you support them? By allowing either tensors or memrefs in the same op?)

Reply:
I wouldn't worry about ViewLikeOpInterface, that is slated to evolve but we only have a very primitive buffer type at the moment.

We can use another name than `view` if you prefer but the key distinction that needs to be visible is whether this copies underlying data or just shuffles pointers around.

Another design possibility if to just call it `columnar_batch` and have a view or copy UnitAttr.

I would avoid cargo culting naming from the bufferization ops as these are exclusively used as "fake casts" during the conversion process.

Reply:
I went with your suggestion in the end. I am thinking of `view` as a verb now: view the operand as a columnar batch, which is a bit different that `ViewLikeOpInterface`, I guess. I was thinking of "reinterpret" but that is longer and more complex to say, so ...